src/commands/power.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+#==============================================================================
+# Copyright (C) 2017 Stephen F. Norledge and Alces Software Ltd.
+#
+# This file/package is part of Alces Metalware.
+#
+# Alces Metalware is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License
+# as published by the Free Software Foundation, either version 3 of
+# the License, or (at your option) any later version.
+#
+# Alces Metalware is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this package.  If not, see <http://www.gnu.org/licenses/>.
+#
+# For more information on the Alces Metalware, please visit:
+# https://github.com/alces-software/metalware
+#==============================================================================
+
+require 'commands/ipmi'
+
+module Metalware
+  module Commands
+    class Power < Ipmi
+      private
+
+      def command
+        hosts = if @options.group
+                  "-g #{@args[0]}"
+                else
+                  "-H #{render_hostname}"
+                end
```

Comment:
In fact both here and in `Console` the `command` methods actually look essentially the same as in `Ipmi` - rather than overriding this and having some duplication between these, would it be possible to just override `render_command` and have that contain just this distinct parts?

For instance in this class I think you would need to move `@args[1].to_s` into `render_command` and then switch on this, and you could then remove the argument to `render_command`; in `Console` you would need to have `render_command` return `-e '&' -I lanplus sol #{type}`.